src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -56,10 +64,24 @@ public IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(IEnumerable<string
                 allExclusionList.UnionWith(privateAssetsExclusionList);
             }
 
-            if(_filterlockFileTarget != null)
+            if(_packagesToBeFiltered != null)
             {
-                IEnumerable<LockFileTargetLibrary> filterLibraries = _filterlockFileTarget.Libraries;
-                Dictionary<string, LockFileTargetLibrary> filterLookup = filterLibraries.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
+                var filterLookup = new Dictionary<string, HashSet<PackageIdentity>>();
+                foreach (var pkg in _packagesToBeFiltered)
+                {
+                    var pkgId = pkg.Id.ToLower();
+                    if (filterLookup.ContainsKey(pkgId))
```

Comment:
Should use "TryGetValue" instead of "ContainsKey + indexing into the dictionary".